src/main/java/com/dietsheet_server/model/diet/Meal.java

```diff
@@ -42,8 +45,21 @@ public Meal() {
     }
 
     public void updateIngredients(List<Ingredient> newIngredients) {
-        this.ingredients.clear();
-        this.ingredients.addAll(newIngredients);
+        ingredients.forEach(ingredient ->
+                ingredient.getProduct().removeIngredientReference(ingredient));
+        ingredients.clear();
+        newIngredients.forEach(ingredient ->
+                ingredient.getProduct().addIngredientReference(ingredient));
+        ingredients.addAll(newIngredients);
```

Comment:
Może zamiast czyszczenia listy i wstawiania do niej nowych elementów lepiej by było po prostu zrobić `ingredients = newIngredients`?

Reply:
Tu chyba jakieś rzeczy z hibernejtem były pamiętam. Żeby poprawnie się wyczyscilo z bazy. Ale ogarnę to jutro jeszcze.

Reply:
https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc